src/clients/credentials/Credentials.tsx

```diff
@@ -72,21 +68,16 @@ export const Credentials = ({ clientId, save, form }: CredentialsProps) => {
   const [open, isOpen] = useState(false);
 
   useFetch(
-    async () => {
-      const providers =
-        await adminClient.authenticationManagement.getClientAuthenticatorProviders();
-
-      const secret = await adminClient.clients.getClientSecret({
-        id: clientId,
-      });
-      return {
-        providers,
-        secret: secret.value!,
-      };
-    },
-    ({ providers, secret }) => {
+    async () =>
+      await Promise.all([
+        adminClient.authenticationManagement.getClientAuthenticatorProviders(),
+        adminClient.clients.getClientSecret({
+          id: clientId,
+        }),
+      ]),
```

Comment:
Nice cleanup, I think you can reduce this even more by removing the `async` keyword and by returning the Promise directly instead of awaiting it.
```suggestion
    () =>
      Promise.all([
        adminClient.authenticationManagement.getClientAuthenticatorProviders(),
        adminClient.clients.getClientSecret({
          id: clientId,
        }),
      ]),
```